diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero3.dts b/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero3.dts
index ec6e70a50327..90c473eb8248 100755
--- a/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero3.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h616-orangepi-zero3.dts
@@ -119,7 +119,7 @@ hdmi_out_con: endpoint {
 
 &gpu {
 	mali-supply = <&reg_dcdc1>;
-	status = "disabled";
+	status = "okay";
 };
 
 &emac0 {
diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
index 2cd542d2a3ea..b88d8e248a30 100755
--- a/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
@@ -184,10 +184,12 @@ gpu: gpu@1800000 {
 			interrupt-names = "job", "mmu", "gpu";
 			clocks = <&ccu CLK_GPU0>, <&ccu CLK_BUS_GPU>;
 			clock-names = "core", "bus";
+			dmas = <&dma 22>, <&dma 22>;
+			dma_names = "rx", "tx";
 			resets = <&ccu RST_BUS_GPU>;
 			//operating-points-v2 = <&gpu_opp_table>;
 			//#cooling-cells = <2>;
-			status = "disabled";
+			status = "okay";
 		};
 
 		video-codec@1c0e000 {
@@ -688,7 +690,7 @@ codec: codec@05096000 {
 			resets = <&ccu RST_BUS_AUDIO_CODEC>;
 			dmas = <&dma 6>;
 			dma-names = "tx";
-			status = "disabled";
+			status = "okay";
 		};
 
 		ahub_dam_plat:ahub_dam_plat@5097000 {
diff --git a/drivers/gpu/drm/drm_gem_dma_helper.c b/drivers/gpu/drm/drm_gem_dma_helper.c
index f6901ff97bbb..1b13f7ce441a 100644
--- a/drivers/gpu/drm/drm_gem_dma_helper.c
+++ b/drivers/gpu/drm/drm_gem_dma_helper.c
@@ -187,7 +187,7 @@ EXPORT_SYMBOL_GPL(drm_gem_dma_create);
  * A struct drm_gem_dma_object * on success or an ERR_PTR()-encoded negative
  * error code on failure.
  */
-static struct drm_gem_dma_object *
+struct drm_gem_dma_object *
 drm_gem_dma_create_with_handle(struct drm_file *file_priv,
 			       struct drm_device *drm, size_t size,
 			       uint32_t *handle)
@@ -215,6 +215,8 @@ drm_gem_dma_create_with_handle(struct drm_file *file_priv,
 	return dma_obj;
 }
 
+EXPORT_SYMBOL_GPL(drm_gem_dma_create_with_handle);
+
 /**
  * drm_gem_dma_free - free resources associated with a DMA GEM object
  * @dma_obj: DMA GEM object to free
diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index 8a3b685c2fcc..19cac237321f 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -2944,6 +2944,43 @@ static const struct panel_desc okaya_rs800480t_7x0gp = {
 	.bus_format = MEDIA_BUS_FMT_RGB666_1X18,
 };
 
+static const struct drm_display_mode olimex_vga_olinuxino_800_mode = {
+	.clock = 40000,
+	.hdisplay = 800,
+	.hsync_start = 800 + 40,
+	.hsync_end = 800 + 40 + 128,
+	.htotal = 800 + 40 + 128 + 88,
+	.vdisplay = 600,
+	.vsync_start = 600 + 5,
+	.vsync_end = 600 + 5 + 4,
+	.vtotal = 600 + 5 + 4 + 19,
+};
+
+static const struct drm_display_mode olimex_vga_olinuxino_1024_mode = {
+	.clock = 45000,
+	.hdisplay = 1024,
+	.hsync_start = 1024 + 16,
+	.hsync_end = 1024 + 16 + 10,
+	.htotal = 1024 + 16 + 10 + 150,
+	.vdisplay = 600,
+	.vsync_start = 600 + 2,
+	.vsync_end = 600 + 2 + 21,
+	.vtotal = 600 + 2 + 21 + 2,
+};
+
+
+static const struct drm_display_mode olimex_lcd_olinuxino_43_mode = {
+	.clock = 12000,
+	.hdisplay = 480,
+	.hsync_start = 480 + 8,
+	.hsync_end = 480 + 8 + 20,
+	.htotal = 480 + 8 + 20 + 23,
+	.vdisplay = 272,
+	.vsync_start = 272 + 4,
+	.vsync_end = 272 + 4 + 10,
+	.vtotal = 272 + 4 + 10 + 13,
+};
+
 static const struct drm_display_mode olimex_lcd_olinuxino_43ts_mode = {
 	.clock = 9000,
 	.hdisplay = 480,
@@ -2956,8 +2993,8 @@ static const struct drm_display_mode olimex_lcd_olinuxino_43ts_mode = {
 	.vtotal = 272 + 8 + 5 + 3,
 };
 
-static const struct panel_desc olimex_lcd_olinuxino_43ts = {
-	.modes = &olimex_lcd_olinuxino_43ts_mode,
+static const struct panel_desc olimex_lcd_olinuxino_43 = {
+	.modes = &olimex_lcd_olinuxino_43_mode,
 	.num_modes = 1,
 	.size = {
 		.width = 95,
@@ -2966,6 +3003,71 @@ static const struct panel_desc olimex_lcd_olinuxino_43ts = {
 	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
 };
 
+static const struct drm_display_mode olimex_lcd_olinuxino_5_mode = {
+	.clock = 33300,
+	.hdisplay = 800,
+	.hsync_start = 800 + 210,
+	.hsync_end = 800 + 210 + 20,
+	.htotal = 800 + 210 + 20 + 26,
+	.vdisplay = 480,
+	.vsync_start = 480 + 22,
+	.vsync_end = 480 + 22 + 10,
+	.vtotal = 480 + 22 + 10 + 13,
+};
+
+static const struct panel_desc olimex_lcd_olinuxino_5 = {
+	.modes = &olimex_lcd_olinuxino_5_mode,
+	.num_modes = 1,
+	.size = {
+		.width = 154,
+		.height = 86,
+	},
+	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
+};
+static const struct drm_display_mode olimex_lcd_olinuxino_7_mode = {
+	.clock = 33300,
+	.hdisplay = 800,
+	.hsync_start = 800 + 210,
+	.hsync_end = 800 + 210 + 20,
+	.htotal = 800 + 210 + 20 + 26,
+	.vdisplay = 480,
+	.vsync_start = 480 + 22,
+	.vsync_end = 480 + 22 + 10,
+	.vtotal = 480 + 22 + 10 + 13,
+};
+
+static const struct panel_desc olimex_lcd_olinuxino_7 = {
+	.modes = &olimex_lcd_olinuxino_7_mode,
+	.num_modes = 1,
+	.size = {
+		.width = 154,
+		.height = 86,
+	},
+	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
+};
+
+static const struct drm_display_mode olimex_lcd_olinuxino_10_mode = {
+	.clock = 45000,
+	.hdisplay = 1024,
+	.hsync_start = 1024 + 16,
+	.hsync_end = 1024 + 16 + 1,
+	.htotal = 1024 + 10 + 6 + 160,
+	.vdisplay = 600,
+	.vsync_start = 600 + 1,
+	.vsync_end = 600 + 1 + 1,
+	.vtotal = 600 + 1 + 1 + 22,
+};
+
+static const struct panel_desc olimex_lcd_olinuxino_10 = {
+	.modes = &olimex_lcd_olinuxino_10_mode,
+	.num_modes = 1,
+	.size = {
+		.width = 222,
+		.height = 143,
+	},
+	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
+};
+
 /*
  * 800x480 CVT. The panel appears to be quite accepting, at least as far as
  * pixel clocks, but this is the timing that was being used in the Adafruit
@@ -4189,8 +4291,23 @@ static const struct of_device_id platform_of_match[] = {
 		.compatible = "okaya,rs800480t-7x0gp",
 		.data = &okaya_rs800480t_7x0gp,
 	}, {
-		.compatible = "olimex,lcd-olinuxino-43-ts",
-		.data = &olimex_lcd_olinuxino_43ts,
+		.compatible = "olimex,olinuxino-vga-800x600",
+		.data = &olimex_vga_olinuxino_800_mode,
+	}, {
+		.compatible = "olimex,olinuxino-vga-1024x768",
+		.data = &olimex_vga_olinuxino_1024_mode,
+	}, {
+		.compatible = "olimex,lcd-olinuxino-4.3",
+		.data = &olimex_lcd_olinuxino_43,
+	}, {
+		.compatible = "olimex,lcd-olinuxino-5",
+		.data = &olimex_lcd_olinuxino_5,
+	}, {
+		.compatible = "olimex,lcd-olinuxino-7",
+		.data = &olimex_lcd_olinuxino_7,
+	}, {
+		.compatible = "olimex,lcd-olinuxino-10",
+		.data = &olimex_lcd_olinuxino_10,
 	}, {
 		.compatible = "ontat,yx700wv03",
 		.data = &ontat_yx700wv03,
diff --git a/drivers/net/wireless/uwe5622/Makefile b/drivers/net/wireless/uwe5622/Makefile
index 313ea512340c..00f41e2b1805 100644
--- a/drivers/net/wireless/uwe5622/Makefile
+++ b/drivers/net/wireless/uwe5622/Makefile
@@ -6,5 +6,5 @@ UNISOCWCN_DIR := $(shell cd $(src)/unisocwcn/ && /bin/pwd)
 UNISOC_BSP_INCLUDE := $(UNISOCWCN_DIR)/include
 export UNISOC_BSP_INCLUDE
 
-UNISOC_FW_PATH_CONFIG := "/lib/firmware/"
+UNISOC_FW_PATH_CONFIG := "/run/current-system/firmware/"
 export UNISOC_FW_PATH_CONFIG
diff --git a/drivers/net/wireless/uwe5622/unisocwcn/platform/rf/rf.c b/drivers/net/wireless/uwe5622/unisocwcn/platform/rf/rf.c
index bfaff8a79a1e..8355dd891d3a 100644
--- a/drivers/net/wireless/uwe5622/unisocwcn/platform/rf/rf.c
+++ b/drivers/net/wireless/uwe5622/unisocwcn/platform/rf/rf.c
@@ -22,14 +22,14 @@
 #include "rf.h"
 #include "wcn_integrate.h"
 
-#define SYSTEM_WIFI_CONFIG_FILE "/lib/firmware/connectivity_configure.ini"
-#define SYSTEM_WIFI_CALI_FILE "/lib/firmware/connectivity_calibration.ini"
+#define SYSTEM_WIFI_CONFIG_FILE "/run/current-system/firmware/connectivity_configure.ini"
+#define SYSTEM_WIFI_CALI_FILE "/run/current-system/firmware/connectivity_calibration.ini"
 #define VENDOR_WIFI_CONFIG_FILE "/vendor/etc/connectivity_configure.ini"
 #define VENDOR_WIFI_CALI_FILE "/vendor/etc/connectivity_calibration.ini"
 #define VENDOR_WIFI_CONFIG_AD_FILE "/vendor/etc/wcn/connectivity_configure.ini"
-#define SYSTEM_WIFI_CONFIG_AD_FILE "/lib/firmware/wcn/connectivity_configure.ini"
-#define VENDOR_WIFI_CALI_AD_FILE "/lib/firmware/wcn/connectivity_calibration.ini"
-#define SYSTEM_WIFI_CALI_AD_FILE "/lib/firmware/wcn/connectivity_calibration.ini"
+#define SYSTEM_WIFI_CONFIG_AD_FILE "/run/current-system/firmware/wcn/connectivity_configure.ini"
+#define VENDOR_WIFI_CALI_AD_FILE "/run/current-system/firmware/wcn/connectivity_calibration.ini"
+#define SYSTEM_WIFI_CALI_AD_FILE "/run/current-system/firmware/wcn/connectivity_calibration.ini"
 #define WIFI_CALI_DUMP_FILE "/var/log/wcn/connectivity_calibration_bak.ini"
 
 #define CONF_TYPE 1
diff --git a/drivers/net/wireless/uwe5622/unisocwcn/platform/wcn_boot.c b/drivers/net/wireless/uwe5622/unisocwcn/platform/wcn_boot.c
index 220bac67ddd0..8e2e770c03c2 100644
--- a/drivers/net/wireless/uwe5622/unisocwcn/platform/wcn_boot.c
+++ b/drivers/net/wireless/uwe5622/unisocwcn/platform/wcn_boot.c
@@ -101,7 +101,7 @@ struct gpio_config {
 #endif
 static char *wcn_fw_path[WCN_FW_MAX_PATH_NUM] = {
 	UNISOC_FW_PATH_DEFAULT,		/* most of projects */
-	"/lib/firmware/"		/* allwinner r328... */
+	"/run/current-system/firmware/" /* allwinner r328... */
 };
 #define WCN_FW_NAME	"wcnmodem.bin"
 #define GNSS_FW_NAME	"gnssmodem.bin"
diff --git a/drivers/net/wireless/uwe5622/unisocwifi/rf_marlin3.c b/drivers/net/wireless/uwe5622/unisocwifi/rf_marlin3.c
index ca307f12ac99..82e4212b788c 100644
--- a/drivers/net/wireless/uwe5622/unisocwifi/rf_marlin3.c
+++ b/drivers/net/wireless/uwe5622/unisocwifi/rf_marlin3.c
@@ -21,7 +21,7 @@
 #ifdef CUSTOMIZE_WIFI_CFG_PATH
 #define WIFI_BOARD_CFG_PATH CUSTOMIZE_WIFI_CFG_PATH
 #else
-#define WIFI_BOARD_CFG_PATH "/lib/firmware"
+#define WIFI_BOARD_CFG_PATH "/run/current-system/firmware"
 #endif
 
 #define CF_TAB(NAME, MEM_OFFSET, TYPE) \
diff --git a/include/drm/drm_gem_dma_helper.h b/include/drm/drm_gem_dma_helper.h
index 8a043235dad8..aeb4d21617cb 100644
--- a/include/drm/drm_gem_dma_helper.h
+++ b/include/drm/drm_gem_dma_helper.h
@@ -32,7 +32,7 @@ struct drm_gem_dma_object {
 #define to_drm_gem_dma_obj(gem_obj) \
 	container_of(gem_obj, struct drm_gem_dma_object, base)
 
-struct drm_gem_dma_object *drm_gem_dma_create(struct drm_device *drm,
+drm_gem_dma_object *drm_gem_dma_create(struct drm_device *drm,
 					      size_t size);
 void drm_gem_dma_free(struct drm_gem_dma_object *dma_obj);
 void drm_gem_dma_print_info(const struct drm_gem_dma_object *dma_obj,
