diff --git a/arch/arm/mach-sunxi/clock_sun50i_h6.c b/arch/arm/mach-sunxi/clock_sun50i_h6.c
index 492fc4a3f..aa27876d1 100644
--- a/arch/arm/mach-sunxi/clock_sun50i_h6.c
+++ b/arch/arm/mach-sunxi/clock_sun50i_h6.c
@@ -23,6 +23,9 @@ void clock_init_safe(void)
 	clrsetbits_le32(&ccm->cpu_axi_cfg, CCM_CPU_AXI_APB_MASK | CCM_CPU_AXI_AXI_MASK,
 			CCM_CPU_AXI_DEFAULT_FACTORS);
 
+	/* enable GPU */
+	writel(0, 0x7010254);
+
 	writel(CCM_PSI_AHB1_AHB2_DEFAULT, &ccm->psi_ahb1_ahb2_cfg);
 	writel(CCM_AHB3_DEFAULT, &ccm->ahb3_cfg);
 	writel(CCM_APB1_DEFAULT, &ccm->apb1_cfg);
